### PR TITLE
Add the meaning of the returned value of PyTypeObject.tp_init

### PR DESCRIPTION
**Pull request by matrixise**

### CODE REVIEW

**Comment by matrixise** [comment]
Should I create an issue and blurb entry for this PR?

**Comment by JulienPalard** [comment]
Hi @matrixise and thanks for this PR.

I don't think we should write that this "Return", as this is just a slot. maybe "The function should return" instead?

Also please double the backticks around the numbers.

**Comment by matrixise** [comment]
In fact, it’s just a copy paste from an other paragraph, but +1 if we need to modify them

File: Doc/c-api/typeobj.rst [hunk 1]
```diff
@@ -1558,6 +1558,8 @@ and :c:type:`PyType_Type` effectively act as defaults.)
    :c:member:`~PyTypeObject.tp_init` function is called; if :c:member:`~PyTypeObject.tp_new` returns an instance of a
    subtype of the original type, the subtype's :c:member:`~PyTypeObject.tp_init` is called.
 
+   Return ``0`` on success, or return ``-1`` and sets an exception on error.
```
Comment (vadmium) [inline Doc/c-api/typeobj.rst]: Change _return_ to _returns_ to match _sets_ (or change _sets_ → _set_)

Comment (matrixise) [reply Doc/c-api/typeobj.rst]: @vadmium done.

File: Doc/c-api/typeobj.rst [hunk 2]
```diff
@@ -1558,6 +1558,8 @@ and :c:type:`PyType_Type` effectively act as defaults.)
    :c:member:`~PyTypeObject.tp_init` function is called; if :c:member:`~PyTypeObject.tp_new` returns an instance of a
    subtype of the original type, the subtype's :c:member:`~PyTypeObject.tp_init` is called.
 
+   Returns ``0`` on success, or return ``-1`` and sets an exception on error.
```
Comment (JulienPalard) [inline Doc/c-api/typeobj.rst]: I'm not native english, but the asymetry (Return**s** / return) may indicate there is something wrong here.
```suggestion
   Returns ``0`` on success, ``-1`` and sets an exception on error.
```

As a non-english speaker I'd rely on grep but hey grep tels we can use return or returns, I'm lost, but mixing both hurt my eyes.

```
$ git grep -oh 'Returns\? ..0...*,.*on' Doc/*/*.rst | sort | uniq -c | sort -gr
      6 Return ``0`` on success, ``-1`` on
      1 Returns ``0`` on success, and ``-1`` on
      1 Returns ``0`` if successful, a nonzero value on
      1 Return ``0`` on success, return ``-1`` on

```

**Review by JulienPalard — APPROVED** [review]
Thanks @matrixise \o/

**Pull request merged**